js_modules/dagster-ui/packages/ui-core/src/asset-graph/sidebar/util.tsx

```diff
@@ -0,0 +1,17 @@
+import {GraphNode} from '../Utils';
+
+export type FolderNodeNonAssetType =
+  | {groupName: string; id: string; level: number}
+  | {locationName: string; id: string; level: number};
+
+export type FolderNodeType = FolderNodeNonAssetType | {path: string; id: string; level: number};
+
+export type TreeNodeType = {level: number; id: string; path: string};
+
+export function nodeId(node: {path: string; id: string} | {id: string}) {
+  return 'path' in node ? node.path : node.id;
```

Comment:
Still think it's not great that a thing with an id has a nodeId which is not the id. Could we rename this nodeKey or something so that `node.id` and `nodeId` are not different.

Reply:
Yeah I can do that